build.cake

```diff
@@ -0,0 +1,243 @@
+    // Clean output directories.
+    CleanDirectories("./src/**/" + configuration);
+    CleanDirectories("./test/**/" + configuration);
+    CleanDirectories("./samples/**/" + configuration);
+  }
+});
+
+Task("Restore-NuGet-Packages")
+  .Description("Restores NuGet packages")
+  .Does(() =>
+{
+  var settings = new DotNetCoreRestoreSettings
+  {
+    Verbose = false,
+    Verbosity = DotNetCoreRestoreVerbosity.Warning
+  };
+  
+  DotNetCoreRestore("./src", settings);
+  DotNetCoreRestore("./test", settings);
+});
+
+Task("Compile")
+  .Description("Builds the solution")
+  .IsDependentOn("Clean")
+  .IsDependentOn("Restore-NuGet-Packages")
+  .Does(() =>
+{
+  var projects = GetFiles("./**/*.xproj");
+  foreach(var project in projects)
+  {
+    DotNetCoreBuild(project.GetDirectory().FullPath, new DotNetCoreBuildSettings {
+      Configuration = configuration,
+      Verbose = false
+    });
+  }
+});
+
+Task("Test")
+  .Description("Executes xUnit tests")
+  .WithCriteria(false)
+  .IsDependentOn("Compile")
+  .Does(() =>
+{
+  var projects = GetFiles("./test/**/*.xproj")
+    - GetFiles("./test/**/Nancy.ViewEngines.Razor.Tests.Models.xproj");
+    
+  foreach(var project in projects)
+  {
+    DotNetCoreTest(project.GetDirectory().FullPath, new DotNetCoreTestSettings {
+      Configuration = configuration
+    });
+  }
+});
+
+Task("Publish")
+  .Description("Gathers output files and copies them to the output folder")
+  .IsDependentOn("Compile")
+  .Does(() =>
+{
+  var files = GetFiles("src/**/bin/" + configuration + "/net452/*.dll")
+    + GetFiles("src/**/bin/" + configuration + "/net452/*.xml")
+    + GetFiles("src/**/bin/" + configuration + "/net452/*.pdb")
+    + GetFiles("src/**/*.ps1");
+
+  CopyFiles(files, outputBinaries);
+});
+
+Task("Package")
+  .Description("Zips up the built binaries for easy distribution")
+  .IsDependentOn("Publish")
+  .Does(() =>
+{
+  var package = outputPackages + File("Nancy-Latest.zip");
+  var files = GetFiles(outputBinaries.Path.FullPath + "/*");
+
+  Zip(outputBinaries, package, files);
+});
+
+Task("Nuke-Symbol-Packages")
+  .Description("Deletes symbol packages")
+  .Does(() =>
+{
+  DeleteFiles(GetFiles("./**/*.Symbols.nupkg"));
+});
+
+Task("Package-NuGet")
+  .Description("Generates NuGet packages for each project that contains a nuspec")
+  .IsDependentOn("Publish")
+  .Does(() =>
+{
+  // Copy the nuspecs to the nuget folder.
+  var nuspecs = GetFiles("./src/**/Nancy*.nuspec");
+  CopyFiles(nuspecs, outputNuGet);
+  nuspecs = GetFiles(outputNuGet.Path.FullPath + "/*.nuspec");
+  
+  // Update all nuspecs.
+  foreach(var nuspec in nuspecs) 
+  {
+    Information("Updating {0}...", nuspec.GetFilename().FullPath);
+    var document = XDocument.Load(nuspec.FullPath);
+    var ns = new XmlNamespaceManager(new NameTable());
+    ns.AddNamespace("ns", "http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd");
+    document.XPathSelectElement("//ns:version", ns).Value = nancyVersion;

```

Comment:
I think we need to make this into a separate tool/method/something, cause it's an insane amount of noise. Does Cake already have something for patching nuspecs?